### ISSUE TITLE
Properly implement alignwide and alignfull styling

### ISSUE THREAD

**Issue by bobbingwide** [bobbingwide/fizzie#44]
It appears I'm not the only one who doesn't really understand how to implement `alignfull` and `alignwide` styling on blocks.

Reading the issue entitled "How to style alignfull and alignwide" ( https://github.com/WordPress/gutenberg/issues/8289 ) I can see a number of solutions but can't see any clearly documented requirements. 

Whichever solution I go for I should understand my own requirements for Fizzie:
- Initially a single column theme with no sidebars
- then extended to support a theme with sidebars.

These are the widths I'm going to use:

Style | Width | Currently used for
------ | ----- | --------------------------
normal | 1140px | most areas except header and footer
alignwide |  1340px | untried 
alignfull | 100% = 1519 pixels on my laptop | header and footer

**Comment by bobbingwide** [comment]
Solution from nicfmc on 15th Jan 2019

After doing much of the same research as most people did here I decided instead of trying to make the wide classes break outside of my wrapper I would add the following to add a max width to blocks without the classes. I have not tested this thoroughly enough yet but it seems to work on columns, images, block quotes. So I just removed the page templates wrapper and assume it as 100%. 

```
.editor-content > [class^="wp-block-"]:not(.alignfull) {
  width: 100%;
  max-width: 1200px;
  margin-left: auto;
margin-right: auto;
}

.editor-content > [class^="wp-block-"]:not(.alignfull).alignwide {
  max-width: 1400px;
}
```
I believe the top class should now be `.wp-site-blocks`

**Comment by bobbingwide** [comment]
This solution works for template-parts. But I can't get images to display wider than the template in which it's contained. 
How to achieve this?

**Comment by bobbingwide** [comment]
I also read https://daverupert.com/2017/03/full-bleed-with-not/ 
to try to get the images to fully go full width when there's a background colour.
I didn't have much success.

**Comment by bobbingwide** [comment]
I also tried using `display: grid` but couldn't make it behave at all. I couldn't work out where to apply the styling.
The CSS was just too hard. 
I appeared to have some luck with the following
but had no idea how to make it apply where I needed it.  

```
grid-template-columns:
        [body-start]
        minmax( 10px, 1fr )
        [wide-start]
        minmax( 0px, 100px )
        [container-start]
        minmax( 300px, 1000px)
        [container-end]
        minmax( 0px, 100px)
        [wide-end]
        minmax( 10px, 1fr )
        [body-end];
```

I appeared to be able to get the header template part to go full width, but had no control over the other areas which I wanted as full width.

**Comment by bobbingwide** [comment]
Following the release of WordPress 5.7 I was prompted to look at how different themes support `alignfull` for the cover block. 
I copied the relevant CSS from TT1-blocks and tried again.  

I created a Full Width cover block using TT1-blocks and viewed the post containing the block as a single post and in the blog archive. Then I attempted to reproduce the result in Fizzie.

Where viewed | TT1 blocks | Fizzie
------- | ------- | -------
Single post | Full width | Full width
Blog page | Wider than a normal blog post | Same as normal blog post

TT1-blocks blog page 
![image](https://user-images.githubusercontent.com/2474435/110833670-9a464800-8294-11eb-8763-049ae3666e91.png)

Fizzie - single.html page
![image](https://user-images.githubusercontent.com/2474435/110839388-64589200-829b-11eb-8130-495e300b7b95.png)

Fizzie - home.html ( blog ) page
![image](https://user-images.githubusercontent.com/2474435/110840421-9d453680-829c-11eb-8b12-a845537d20ce.png)

**Comment by bobbingwide** [comment]
There's a new way of doing this alignfull stuff.

As documented by Anne McCarthy in #fse-outreach-experiment

> TLDR: a new PR landed that reimagines alignments with block themes. This change first landed in Gutenberg in 10.3 and then in TT1 Blocks. Both have new versions today so please update to the latest when you’re testing. If you update Gutenberg but not TT1 Blocks, the editor will show up strangely as pictured below with everything shoved to one side. If you are a theme author and curious to learn more, check out what TT1 blocks did here as well as guidance given to block authors here.

I've already tried it in https://github.com/bobbingwide/sb/issues/4

I need to apply similar changes to Fizzie. The changes should include:
- [ ] updating the template parts that include unnecessary HTML. See below.
- [ ] reworking my override of `template-part.php` to support the `area` attribute. See SB theme
- [ ] Updating the stylesheet to cater for different HTML for header and footer areas.

Unnecessary HTML includes:

Group block - 13 occurrences
```
<div class="wp-block-group__inner-container">
```

**Comment by bobbingwide** [comment]
Once again I'm completely baffled. 
In the Block settings for my Header template part the Layout panel just displays "Customize the width for all elements that are assigned to the center or wide columns."


In a Group block within a page template I see the Content and Wide entry fields.

![image](https://user-images.githubusercontent.com/2474435/114050794-7c183b80-9884-11eb-9d8e-4efbee8ab53d.png)

These also appear for other template parts.

**Comment by bobbingwide** [comment]
Additionally, when inspecting the front end output, I've got unwanted paragraph tags in the inline styles setting the `.wp-container-hhhhhhhhhh` styles. 
![image](https://user-images.githubusercontent.com/2474435/114063475-8b9d8180-9890-11eb-910c-8c34b7a246a5.png)

**Comment by bobbingwide** [comment]
> Once again I'm completely baffled.

The solution that appears to work, is to edit the templates and template parts .html files
adding `"align":"wide"` or `"align":"full"` and either `"layout":{"inherit":true}` or `"layout":{"contentSize":"80%","wideSize":"90%"}`

**Comment by bobbingwide** [comment]
| Once again I'm completely baffled

What I didn't mention was that I never saw the **Inherit default layout** toggle.
Once again, this was due to a syntax error in `experimental-theme.json`, a missing close curly brace.

I've now got the toggle to display...
![image](https://user-images.githubusercontent.com/2474435/114143487-225b5400-990c-11eb-89cb-0c06f8a24ae1.png)

But I still don't understand how to use the controls. 
Nor do I understand when the toolbar displays the align wide and align full icons.

I see them in Fizzie but not in TT1 blocks.

**Comment by bobbingwide** [comment]
I've been making progress as you'll see by a few commits soon. But I still didn't know how I could get my images to be aligned wide and full. I thought that I'd need to use a custom template, but couldn't see how to do this.  I asked in Slack 

> Q. How do you create "page templates" with FSE.

And Carolina Nymark gave me the link to the documentation I needed. 
https://developer.wordpress.org/block-editor/how-to-guides/themes/theme-json/#other-theme-metadata

Armed with that information I set about defining a custom template.


1. Add the definition to `experimental-theme.json`.
2. Remove the comments from the example so that json_decode works.
2. Create the new block template file `single-full.html`; copied from `single.html`
3. Fiddle about with the template part `post-content.html`
4. Write some new content to test it.
5. Suffer #62 along the way.
6. Do a couple of screen captures to show that it works with 10.3.1

```
"customTemplates": [
    {
      "name": "single-full",
      "title": "Full width stuff",
      "postTypes": [
        "page",
        "post",
        "block",
        "block_example"
      ]
    }
  ]
```

This shows the Post Attributes Settings Panel where I've chosen the Template.
In Status and Visibility the Template is single-full.
![image](https://user-images.githubusercontent.com/2474435/114368963-d4e21f80-9b75-11eb-8751-0790ef1181d1.png)

Here we can see the results on the Front End.
I've zoomed right out so that you can see the three images: normal, wide and full width.
Interestingly the full width image doesn't scale down.

![image](https://user-images.githubusercontent.com/2474435/114369218-11158000-9b76-11eb-908e-8159bf97a738.png)

I haven't yet tried to create content with other blocks being wide and full.

**Comment by bobbingwide** [comment]
I changed the height of my test image so that I don't have to scroll to see all three. In the post version using the custom template ( single-full.html ) the images look OK. 
![image](https://user-images.githubusercontent.com/2474435/114383402-6907b300-9b85-11eb-9939-da349cd15751.png)

But in the block_example equivalent, where I haven't (yet) wrapped everything in a group, the wide image is a bit wonky. 
It appears to be full width aligned left. 
![image](https://user-images.githubusercontent.com/2474435/114383649-aff5a880-9b85-11eb-9975-24b6d4330e64.png)

**Comment by bobbingwide** [comment]
And I get a different result when the post type is page.
Normal is left aligned and wide is full width.
![image](https://user-images.githubusercontent.com/2474435/114386014-9ace4900-9b88-11eb-931e-35539988606a.png)

### Solution / workaround

Convert the content to a group block and toggle Inherit default layout on.

**Comment by bobbingwide** [comment]
I think I'm getting the hang of this at last.

**Comment by bobbingwide** [comment]
Well, that's what I thought. Footer's gone tits up again with GB 10.4.0
![image](https://user-images.githubusercontent.com/2474435/114739786-196de680-9d41-11eb-8891-eb4fff0df50a.png)

**Comment by bobbingwide** [comment]
Changing the HTML for the template part to add `"inherit":` true alters the footer so that the background gradient is applied only to the wide bit.  No screen capture for this intermediate stage.
`<!-- wp:template-part {"slug":"page-footer","theme":"fizzie", "align":"full","layout":{"inherit": true }} /-->`

**Comment by bobbingwide** [comment]
I believe the Site Logo size issue is associated with https://github.com/WordPress/gutenberg/pull/30843 or https://github.com/WordPress/gutenberg/pull/30845

**Comment by bobbingwide** [comment]
> There's another tiny problem. Every now and then I see small bits of styling of empty paragraphs in the third block column. It happens with the last paragraph after the `<div class="bw_blog "><a href="https://blocks.wp.a2z/">WordPress Blocks</a>`</div>

See also #20

**Comment by bobbingwide** [comment]
Delivered in v0.5.0
